app/javascript/dashboard/routes/dashboard/settings/attributes/EditAttribute.vue

```diff
@@ -0,0 +1,205 @@
+        </woot-button>
+        <woot-button variant="clear" @click.prevent="onClose">
+          {{ $t('ATTRIBUTES_MGMT.ADD.CANCEL_BUTTON_TEXT') }}
+        </woot-button>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+import { required, minLength } from 'vuelidate/lib/validators';
+import alertMixin from 'shared/mixins/alertMixin';
+export default {
+  components: {},
+  mixins: [alertMixin],
+  props: {
+    selectedAttribute: {
+      type: Object,
+      default: () => {},
+    },
+  },
+  data() {
+    return {
+      displayName: '',
+      description: '',
+      attributeModel: 0,
+      attributeType: 0,
+      models: [
+        {
+          id: 0,
+          option: 'Conversation',
+        },
+        {
+          id: 1,
+          option: 'Contact',
+        },
+      ],
+      types: [
+        {
+          id: 0,
+          option: 'Text',
+        },
+        {
+          id: 1,
+          option: 'Number',
+        },
+        {
+          id: 2,
+          option: 'Currency',
+        },
+        {
+          id: 3,
+          option: 'Percent',
+        },
+        {
+          id: 4,
+          option: 'Link',
+        },
+        {
+          id: 5,
+          option: 'Date',
+        },
+      ],
+      show: true,
+    };
+  },
+  validations: {
+    displayName: {
+      required,
+    },
+    attributeType: {
+      required,
+    },
+    description: {
+      required,
+      minLength: minLength(1),
+    },
+    attributeModel: {
+      required,
+    },
+  },
+  computed: {
+    ...mapGetters({
+      uiFlags: 'attributes/getUIFlags',
+    }),
+    pageTitle() {
+      return `${this.$t('ATTRIBUTES_MGMT.EDIT.TITLE')} - ${
+        this.selectedAttribute.attribute_display_name
+      }`;
+    },
+    attributeKey() {
+      return this.displayName
+        .toLowerCase()
+        .replace(/[^\w ]+/g, '')
+        .replace(/ +/g, '_');
+    },
+  },
+  mounted() {
+    this.setFormValues();
+  },
+  methods: {
+    onClose() {
+      this.$emit('on-close');
+    },
+    setFormValues() {
+      this.displayName = this.selectedAttribute.attribute_display_name;
+      this.description = this.selectedAttribute.attribute_description;
+    },
+    async editAttributes() {
+      this.$v.$touch();
+      if (this.$v.$invalid) {
+        return;
+      }
+      try {
+        await this.$store.dispatch('attributes/update', {
+          id: this.selectedAttribute.id,
+          attribute_display_name: this.displayName,
+          attribute_description: this.description,
+          attribute_model: this.attributeModel,
+          attribute_display_type: this.attributeType,
+          attribute_key: this.attributeKey,
+        });
+        this.showAlert(this.$t('ATTRIBUTES_MGMT.EDIT.API.SUCCESS_MESSAGE'));
+        this.onClose();
+      } catch (error) {
+        this.showAlert(this.$t('ATTRIBUTES_MGMT.EDIT.API.ERROR_MESSAGE'));
```

Comment:
Check any message exist in the `error` object. Otherways show the default error message